### PR TITLE
obetsun blue hat solution

### PR DESCRIPTION
**Pull request by olegbet**
obetsun blue hat solution

### CODE REVIEW

File: blue-hat/obetsun/blue-hat.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+	"image"
+	"image/color"
+	"image/png"
+	"os"
+)
+
+func main() {
+	f, err := os.Open("red_hat.png")
+	if err != nil {
+		panic(err)
```
Comment (davidmogar) [inline blue-hat/obetsun/blue-hat.go:13]: This prints the stack trace. Not sure if that's what you were going for.

File: blue-hat/obetsun/blue-hat.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+	}
+	defer f.Close()
+
+	src, _, err := image.Decode(f)
+	if err != nil {
+		panic(err)
+	}
+
+	bounds := src.Bounds()
+	w, h := bounds.Max.X, bounds.Max.Y
```
Comment (davidmogar) [inline blue-hat/obetsun/blue-hat.go:23]: This can be moved to the for loop as you only used them there.

File: blue-hat/obetsun/blue-hat.go [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+	blue := image.NewRGBA(bounds)
+
+	for x := 0; x < w; x++ {
+		for y := 0; y < h; y++ {
+			oldColor := src.At(x, y)
+			r, g, b, a := oldColor.RGBA()
+
+			if r > g+b {
+				blue.SetRGBA(x, y, color.RGBA{
```
Comment (davidmogar) [inline blue-hat/obetsun/blue-hat.go:32]: You only need this to generate the blue hat. The if is not needed.

File: blue-hat/obetsun/blue-hat.go [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+
+	r := image.Rect(0, 0, width, height)
+	img := image.NewRGBA(r)
+
+	//Fill name background with white color
+	for i := 0; i < width; i++ {
+		for j := 0; j < height; j++ {
+			img.Set(i, j, color.RGBA{255, 255, 255, 255})
+		}
+	}
```
Comment (hemartin-at-redhat) [inline blue-hat/obetsun/blue-hat.go:111]: You can do it all at once: 

```
m := image.NewRGBA(image.Rect(0, 0, 640, 480))
blue := color.RGBA{0, 0, 255, 255}
draw.Draw(m, m.Bounds(), &image.Uniform{blue}, image.ZP, draw.Src)
```

From https://go.dev/blog/image-draw

File: blue-hat/obetsun/blue-hat.go [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+
+	//One pixel border around the name
+	offsetCol := 1
+	offsetRow := 1
+
+	//Draw scaled name
+	for i, rn := range name {
+
+		var letter = m[string(rn)]
+		for j, p := range letter {
```
Comment (hemartin-at-redhat) [inline blue-hat/obetsun/blue-hat.go:121]: If `j` is not related with `i`, I would not name it this way. It is common to have `i` and `j` as indexes for two-dimensional matrices. Here the `j` can make someone think it is related with the `i`.

File: blue-hat/obetsun/blue-hat.go [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+	//Fill name background with white color
+	for i := 0; i < width; i++ {
+		for j := 0; j < height; j++ {
+			img.Set(i, j, color.RGBA{255, 255, 255, 255})
+		}
+	}
+
+	//One pixel border around the name
+	offsetCol := 1
+	offsetRow := 1
```
Comment (hemartin-at-redhat) [inline blue-hat/obetsun/blue-hat.go:115]: Those two can be moved to the top as constants I think.

**Review by hemartin-at-redhat — COMMENTED** [review]
:+1:

File: blue-hat/obetsun/blue-hat.go (hunk 6 above)
Comment (olegbet) [reply blue-hat/obetsun/blue-hat.go:115]: offsetCol - yes, but offsetRow is being changed during the loop

**Pull request merged**